File: mvc1/Python/11723 집합.py
import sys

num = int(sys.stdin.readline())
arr = []
for _ in range(num) :
    order = sys.stdin.readline().split()

    if order[0] == "add" :
        x = int(order[1])
        if x not in arr : arr.append(x)
    elif order[0] == "remove" :
        x = int(order[1])
        if x in arr : arr.remove(x)
    elif order[0] == "check" :
        x = int(order[1])
        if x in arr : print(1)
        else : print(0)
    elif order[0] == "toggle" :
        x = int(order[1])
        if x in arr : arr.remove(x)
        else : arr.append(x)
    elif order[0] == "all" : arr = [i+1 for i in range(20)]
    elif order[0] == "empty" : arr = []

File: _etc/C#/11723 집합.cs
using System;
using System.Text;
using static System.Console;
using System.Collections.Generic;




class Set {
    List<int> set = new List<int>();
    public StringBuilder sb = new StringBuilder();
    
    public void Add(int x) {
        if(!set.Contains(x)) set.Add(x);
    }
    public void Remove(int x) {
        if(set.Contains(x)) set.Remove(x);
    }
    public void Check(int x) {
        if(set.Contains(x)) sb.Append("1\n");
        else sb.Append("0\n");
    }
    public void Toggle(int x) {
        if(set.Contains(x)) set.Remove(x);
        else set.Add(x);
    }
    public void All() {
        set.Clear();
        for(int i = 1; i <= 20; i++) {
            set.Add(i);
        }
    }
    public void Empty() {
        set.Clear();
    }
    
}

class Program
{
    static void Main() {
        Set st = new Set();
        string s = ReadLine();
        int m = int.Parse(s);
        for(int i = 0 ; i < m; i++) {
            string[] order = ReadLine().Split(' ');
            
            
            if(order[0] == "add") {
                st.Add(int.Parse(order[1]));
            }
            else if(order[0] == "check") {
                st.Check(int.Parse(order[1]));
            }
            else if(order[0] == "remove") {
                st.Remove(int.Parse(order[1]));
            }
            else if(order[0] == "toggle") {
                st.Toggle(int.Parse(order[1]));
            }
            else if(order[0] == "all") {
                st.All();
            }
            else if(order[0] == "empty") {
                st.Empty();
            }
        }
        
        WriteLine(st.sb);
    }
}
